src/components/Attachments/AttachmentCarousel/extractAttachmentsFromReport.js

```diff
@@ -35,7 +42,17 @@ function extractAttachmentsFromReport(parentReportAction, reportActions) {
             if (name === 'img' && attribs.src) {
                 const expensifySource = attribs[CONST.ATTACHMENT_SOURCE_ATTRIBUTE];
                 const source = tryResolveUrlFromApiRoot(expensifySource || attribs.src);
-                const fileName = attribs[CONST.ATTACHMENT_ORIGINAL_FILENAME_ATTRIBUTE] || FileUtils.getFileName(`${source}`);
+                if (uniqueSources.has(source)) {
+                    return;
+                }
+
+                uniqueSources.add(source);
+                let fileName = attribs[CONST.ATTACHMENT_ORIGINAL_FILENAME_ATTRIBUTE] || FileUtils.getFileName(`${source}`);
+
+                // image.jpg helps render the attachment as image if we didn't obtain an extension from the source
+                if (!/\.\w+$/.test(fileName)) {
```

Comment:
**suggestion:** put this into a `CONST.REGEX` so it is clear we are checking that it is a file.

Reply:
Hey, thanks for the suggestion! I see where you're coming from with the idea to move the pattern to a `CONST.REGEX`. However, I think it's best not to go down that route right now. The pattern isn't suitable for general use, so I'd like to prevent it becoming a permanent part of CONST.

I'm preparing a list of follow-up actions. The goal here is to either find a way to refactor the code, or at least spark a discussion on the matter.

Reply:
I've updated the code to use a method from `FileUtils` and removed the regex pattern.

Reply:
Thanks! My original concern is avoiding having to mentally parse the regex when reading through the code so a utils method also solves it without moving it to `CONST`.